terrascan.sh

```diff
@@ -47,11 +51,50 @@ function common::parse_cmdline {
   done
 }
 
+function common::is_hook_run_on_whole_repo {
+  local -a -r files=("$@")
+  local sorted_string
+  local git_ls_files
+  local included_files
+  local SCRIPT_DIR
+
+  # get directory containing this script
+  SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+  # get included and excluded files from .pre-commit-hooks.yaml file
+  included_files=$(sed -n "/^- id: $HOOK_ID$/,/^$/p" "$SCRIPT_DIR/.pre-commit-hooks.yaml" | awk '$1 == "files:" {print $2; exit}')
+  excluded_files=$(sed -n "/^- id: $HOOK_ID$/,/^$/p" "$SCRIPT_DIR/.pre-commit-hooks.yaml" | awk '$1 == "exclude:" {print $2; exit}')
+
+  # sorted string with the files passed to the hook by pre-commit
+  sorted_string=$(printf '%s\n' "${files[@]}" | sort | tr '\n' ' ')
+
+  # git ls-files sorted string
+  if [[ "$excluded_files" != "" ]]; then
+    git_ls_files=$(git ls-files | sort | grep -v "$excluded_files" | grep -e "$included_files" | tr '\n' ' ')
```

Comment:
I'm not quite sure how `pre-commit` interprets `files` and `exclude` parameters passed altogether but from what I lean to think it's something like "include `files` AND exclude `exclude`" which means that we need to reproduce this particular order for the sake of sameness:
```suggestion
    git_ls_files=$(git ls-files | sort | grep "$included_files" | grep -v "$excluded_files" | tr '\n' ' ')
```
What do you think?

Reply:
`excluded_files` can be empty. `grep -v ""` just excludes all

Reply:
Well, right above this line there a respective condition `if [[ "$excluded_files" != "" ]]; then` to protect from empty inverse `grep`.
My suggestion was to obey the ordering of inclusion and exclusion and have the order be the same as `pre-commit`'s logic has (I may be wrong assuming that `pre-commit` applies inclusion before exclusion though).

Reply:
Ah, my bad. Nevermind

Reply:
> My suggestion was to obey the ordering of inclusion and exclusion and have the order be the same as pre-commit's logic has (I may be wrong assuming that pre-commit applies inclusion before exclusion though).

@yermulnik I was reading the [pre-commit documentation](https://pre-commit.com/#:~:text=of%20files%20to%20run%20on.-,exclude,(optional%3A%20default%20%5E%24)%20exclude%20files%20that%20were%20matched%20by%20files.,-types) and I understand that `pre-commit` first include the `files` and then exclude files that were matched by `files`. You are right!!